kolibri/core/content/api.py

```diff
@@ -1310,6 +1315,79 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(ValuesViewset):
+    pagination_class = OptionalPageNumberPagination
+
+    queryset = ContentDownloadRequest.objects.all()
+
+    field_map = {
+        "source_id": "self.request.user.id",
+    }
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+    )
+
+    def filter_queryset(self, queryset):
```

Comment:
Note that this is overwriting the base `filter_queryset` method which is what applies FilterSets to the queryset: https://github.com/encode/django-rest-framework/blob/38a74b42da10576857d6bf8bd82a73b15d12a7ed/rest_framework/generics.py#L144

It's better to do this kind of global filtering of querysets for a viewset in the `get_queryset` method, so something like:
```
def get_queryset(self):
    return ContentDownloadRequest.objects.filter(source_id=self.request.user_id)
```

This also means you can dispense with the `queryset` attribute on the class, as `get_queryset` will override this.

Reply:
This should be updated still.